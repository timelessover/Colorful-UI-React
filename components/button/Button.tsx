import * as React from 'react'
import { ButtonProps } from 'types/button'
import cx from 'classnames'
import { Wave } from '../utils'
import Icon from '../icon/Icon'
import './style'

const componentName = 'cl-button'

const defaultProps: ButtonProps = {
  cn:componentName,
  position: 'left',
  size: 'default',
  type: 'primary',
  htmlType: 'button',
  loading: false,
  disabled: false,
  onClick: () => { }
}

function Button(userProps: ButtonProps, ref: any) {
  const props = {
    ...defaultProps,
    ...userProps
  }
  const renderIcon = () => {
    const { icon, size, loading } = props
    const className = cx('icon', [size], { loading })
    return loading ? (
      <Icon name="loading" className={className} />
    ) : (
        icon && <Icon name={icon} className={className} />
      )
  }
  const {
    cn,
    position,
    icon,
    loading,
    size,
    type,
    htmlType,
    style,
    className,
    disabled,
    children,
    ...rest
  } = props

  const buttonClassName = cx(cn, [position, size, type, className], {
    disabled
  })

  return (
    <Wave>
      <button
        className={buttonClassName}
        style={style}
        type={htmlType}
        disabled={disabled}
        {...rest}
      >
        {renderIcon()}
        <span className={cx(`${cn}-inner`)}>{children}</span>
      </button>
    </Wave>
  )
}




export default Button
