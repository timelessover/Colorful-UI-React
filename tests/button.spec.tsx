import * as React from 'react'
import { render } from '@testing-library/react'
import Button from '../components/button/Button'

describe('render button', () => {
  it('render default', () => {
    const wrapper = render(<Button/>)
    // console.log(wrapper.container)
    expect(wrapper.container).toMatchSnapshot()
  })
 }
)
