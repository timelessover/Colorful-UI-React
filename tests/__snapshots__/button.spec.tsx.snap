// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button render default 1`] = `
<div>
  <button
    class="cl-button left default primary"
    type="button"
  >
    <span
      class="cl-button-inner"
    />
  </button>
</div>
`;
